### PR TITLE
Task6/Routing

### PR DESCRIPTION
**Pull request by OMManko**

### CODE REVIEW

File: src/components/movieInfo/movieInfo.jsx [hunk 1]
```diff
@@ -4,24 +4,57 @@ import "./movieInfo.scss";
 import PropTypes from "prop-types";
 import MoviePoster from "../moviePoster/moviePoster";
 import { moviePosterSizes } from '../../constants/constants';
+import { connect } from "react-redux";
+import { fetchMovie } from "../../actions/actions";
+import { withRouter } from "react-router-dom";
+
+class MovieInfo extends React.Component {
+    componentDidMount () {
+        const params = +this.props.match.params.id;
```
Comment (nsasinovich) [inline src/components/movieInfo/movieInfo.jsx]: the name of the variable is a bit confusing, it's rather id than params

File: src/components/searchForm/searchForm.jsx [hunk 2]
```diff
@@ -1,48 +1,88 @@
...
+            },
+            {
+                id: filterOptions.GENRE,
+                label: "genre",
+                checked: selectedFilterOption === filterOptions.GENRE
+            }
+        ];
+
+        const handleSearchMovies = () => {
+            this.props.history.push(`/search?search=${inputValue}&searchBy=${selectedFilterOption}`);
```
Comment (nsasinovich) [inline src/components/searchForm/searchForm.jsx]: add to distructuring above

File: src/components/searchForm/searchForm.jsx [hunk 3]
```diff
@@ -1,48 +1,88 @@
...
+
+const mapStateToProps = state => ({
+    selectedFilterOption: state.selectedFilterOption,
+    inputValue: state.searchInputValue
+});
+
+const mapDispatchToProps = {
+    changeCurrentSearchOption: changeSearchOption,
+    searchMovies: fetchMovies,
+    updateInputValue: updateInputValue
```
Comment (nsasinovich) [inline src/components/searchForm/searchForm.jsx]: can be simply `updateInputValue`

File: src/components/searchResults/searchResults.jsx [hunk 4]
```diff
@@ -3,65 +3,52 @@ import { hot } from "react-hot-loader";
...
+            selectedSortOption
+        } = this.props;
+
+
+        const formattedMoviesList = moviesList.map((movie) => {
+            movie.release = new Date(movie.release_date).getUTCFullYear();
+            return movie;
+        });
+
+        moviesList.sort((option1, option2) => option2[selectedSortOption] - option1[selectedSortOption]);
```
Comment (nsasinovich) [inline src/components/searchResults/searchResults.jsx]: props are read-only, do not modify them, please

File: src/components/movieInfo/movieInfo.jsx (hunk 1 above)
Comment (OMManko) [reply src/components/movieInfo/movieInfo.jsx]: changed

File: src/components/searchForm/searchForm.jsx (hunk 2 above)
Comment (OMManko) [reply src/components/searchForm/searchForm.jsx]: done

File: src/components/searchResults/searchResults.jsx (hunk 4 above)
Comment (OMManko) [reply src/components/searchResults/searchResults.jsx]: done

File: src/components/searchForm/searchForm.jsx (hunk 3 above)
Comment (OMManko) [reply src/components/searchForm/searchForm.jsx]: done

**Pull request merged**